code-challenges.test.js

```diff
@@ -12,40 +12,179 @@
+    // Expected output: -31
+
+    expect(endTally(votes1)).toEqual(11);
+    expect(endTally(votes2)).toEqual(-31);
+  });
+});
+// --->  FAIL  ./code-challenges.test.js
+//   endTally
+//   ✕ takes in an object that contains up votes and down votes and returns the end tally. (1 ms)
+
+// ● endTally › takes in an object that contains up votes and down votes and returns the end tally.
+
+//   ReferenceError: endTally is not defined
 
 // b) Create the function that makes the test pass.
 
+const votes1 = { upVotes: 13, downVotes: 2 };
+const votes2 = { upVotes: 2, downVotes: 33 };
+
+const endTally = (object) => {
+  const addItUp = (number) => {
+    let votes = number - number;
+  };
+  return endTally;
+};
+console.log(votes1);
+console.log(votes2);
+
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
-// a) Create a test with an expect statement using the variables provided.
+// psuedocode: Create a function named noDups that takes in two arrays as arguments and returns one array with no duplicate values.
+// Input: 2 Arrays
+// Output: 1 Array
+// create function called noDups that takes in two arrays as arguments and returns one array with no duplicate values.
+// I think I can join the two Arrays with .join
+// must fine a methed to remove duplicates. Maybe .filter.
+
+// a) Create a test with expect statements for each of the variables provided.
 
-const dataArray1 = ["array", "object", "number", "string", "Boolean"]
-const dataArray2 = ["string", "null", "Boolean", "string", "undefined"]
-// Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
+describe("noDups", () => {
+  it("takes in two arrays as arguments and returns one array with no duplicate values.", () => {
+    const dataArray1 = ["array", "object", "number", "string", "Boolean"];
+    const dataArray2 = ["string", "null", "Boolean", "string", "undefined"];
+    // Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
+
+    expect(noDups(dataArray1, dataArray2)).toEqual([
+      "array",
+      "object",
+      "number",
+      "string",
+      "Boolean",
+      "null",
+      "undefined",
+    ]);
+  });
+});
+
+// --->  FAIL  ./code-challenges.test.js
+//   noDups
+//   ✕ takes in two arrays as arguments and returns one array with no duplicate values.
+
+// ● noDups › takes in two arrays as arguments and returns one array with no duplicate values.
+
+//   ReferenceError: noDups is not defined
 
 // b) Create the function that makes the test pass.
+
+const dataArray1 = ["array", "object", "number", "string", "Boolean"];
+const dataArray2 = ["string", "null", "Boolean", "string", "undefined"];
+
+const noDups = (array1, array2) => {
+  const newArray1 = [...new Set(string)];
+  for (let i = 0; i < array.length; i++) {
```

Comment:
Here you are using variables that do not exist: `string` and `array`, which will cause an error.